gb-cpu/src/microcode/math.rs

```diff
@@ -67,6 +67,60 @@ fn test_sub_components() {
     )
 }
 
+pub fn sub_components_u16(a: u16, b: u16) -> (u16, Flag) {
+    let (res, overflowing) = a.overflowing_sub(b);
+    (
+        res,
+        Flag {
+            half_carry: (a & 0xff) < (b & 0xff),
```

Comment:
After some though, the half carry is use for for **decimal adjust**.
on the Gameboy only the opcode `daa` is natively present and only work on register `A`.
we should confirm how the `half carry` work on the opcode `Add SP, i8`

Reply:
Should I put the PR in draft until then ? Or did I misunderstood you.

Reply:
not need, we could check later on